### PR TITLE
PIN, Duress PIN, and Duress Passphrase

### PR DESCRIPTION
**Pull request by deregs**
Relates to issues: #857, #859 
Features added for password/duress password:
- Set/Change Duress Password screen added.
- If a password is set, user now has to enter their password in order to modify password/duress password
- Delete button added to set password and set duress password screens, only appears if the password/duress password is already set.
- If duress password entered while logging in, all node configs are deleted and user is logged in.

Features added for PIN/duress PIN
- Set/Change PIN and Set/Change Duress PIN screens added.
- Set/Change PIN/Duress PIN screens allow PINs of length 4-8 to be set.
- Any time the Pin Pad is displayed, the order of the numbers is shuffled and none of the numbers visually respond to touch events - this is to prevent against attacks.
- If PIN is set, user has to enter PIN in order to modify PIN/Duress PIN.
- Delete options added in Security settings for PIN/Duress PIN, each one appears only when the corresponding PIN/Duress PIN has already been set.
- If duress PIN is entered while logging in, all node configs are deleted and user is logged in.

Other notes:
- PinPad component can be reused on the SendReceive pane with numbers not shuffled.
- PIN and Password can not be set at the same time - must choose one or the other. But user can switch from one to the other by deleting PIN/Password and then setting the other in Security settings.
- User is restricted to 4 failed authentication attempts in a row, and failed attempts are saved even if the app is closed. When the user submits a 5th failed attempt, app takes duress behavior.
- Security Settings will show different options depending on what values are set:
  - If no Password or PIN is set, shows options to set PIN or set Password
  - If Password is set, shows options to set/change Password or set/change Duress Password
  - If PIN is set, shows options to set/change PIN, Delete PIN, or set/change Duress PIN
  - If PIN and Duress PIN are set, shows all the PIN options plus option to Delete Duress Password

pin

### CODE REVIEW

**Comment by Bosch-0** [comment]
> Should duress password be cleared when password is changed?

I'd say no as well. This is an awesome feature! ty for this deregs. Will be tidying up the UI frames this week and will make sure this is integrated cleanly

**Comment by Bosch-0** [comment]
Closes https://github.com/ZeusLN/zeus/issues/857 https://github.com/ZeusLN/zeus/issues/859

File: components/Pin.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+export default function Pin({
+    onSubmit,
+    onPinChange = () => void 0,
+    hidePinLength,
+    pinLength = 4,
+    pinConfirm = false
+}: PinProps) {
+    const [pinValue, setPinValue] = useState('');
```
Comment (kaloudis) [inline components/Pin.tsx:21]: love the use of hooks. we should use them wherever we can.

File: components/PinCircles.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            margin: 10
+        }
+    });
+
+    let filled = [];
+    for (let i = 0; i < numFilled; i++) {
+        filled.push(<Filled style={styles.circles} />);
+    }
+
+    let hollow = [];
```
Comment (kaloudis) [inline components/PinCircles.tsx:32]: nit: `filled` and `hollow` can be consts.

File: package.json [hunk 3]
```diff
@@ -112,8 +112,8 @@
     "@types/jest": "23.3.13",
     "@types/lodash": "4.14.120",
     "@types/object-hash": "2.2.1",
-    "@types/react": "16.9.19",
-    "@types/react-native": "0.61.4",
+    "@types/react": "17.0.2",
+    "@types/react-native": "0.67.2",
```
Comment (kaloudis) [inline package.json:116]: good fix

File: components/PinCircles.tsx (hunk 2 above)
Comment (deregs) [reply components/PinCircles.tsx]: done

**Pull request merged**